### PR TITLE
New version of SYMPHONY

### PR DESCRIPTION
**Pull request by tkralphs**

### CODE REVIEW

**Comment by dpo** [comment]
Remove the revision number when you upgrade the version. Bump it up when you fix the formula in some other way.

File: symphony.rb [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
 
   head "https://github.com/coin-or/SYMPHONY"
 
-  bottle do
-    root_url "https://github.com/coin-or-tools/homebrew-coinor/releases/download/symphony-5.6.17_1"
-    sha256 cellar: :any,                 big_sur:      "1abc5b95dd71c4f8adf52c2e3d4dd7fb0ca3ef603919682c2bc4057ecec43e4a"
-    sha256 cellar: :any,                 catalina:     "b96391eb3099afa52d40d7edf30f72ba98857483a3c67f523c37f331ed6a7987"
-    sha256 cellar: :any_skip_relocation, x86_64_linux: "7597c630fadef7c6ba99197d16f9a00d0c2f334cf04b757cc1f431e7e4977652"
-  end
-
```
Comment (dpo) [inline symphony.rb:16]: no need to remove this; it will be updated

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by tkralphs** [comment]
@dpo I'm not sure I know what I'm doing here, but took a shot. I couldn't update the recipe locally without deleting the `bottle do` part, which had a hunch of SHAs in it, which would need to be updated anyway. I assume those are somehow added automagically in the build process here?

**Comment by dpo** [comment]
There's no need to touch the `bottle` section ; it will be automatically updated when the PR is merged with the `pr-pull` label.

**Comment by tkralphs** [comment]
The only reason I removed that section was because otherwise, if I just change the source URL to the new version, the whole build just fails immediately without even trying to build from source. It doesn't give me the new SHA either. I'm probably using the wrong workflow. If I just change the source URL and then push without testing locally, it seems like I will just run into the same issue withe automatic build.

**Comment by dpo** [comment]
Try removing the `revision`

**Comment by tkralphs** [comment]
Even if I remove the `revision`, if I change the source URL and try to either reinstall or install from scratch, I get:
```
==> Downloading https://github.com/coin-or-tools/homebrew-coinor/releases/download/symphony-5.6.18/symphony-5.6.18.x86_64_linux.bottle.tar.gz
curl: (22) The requested URL returned error: 404

Error: symphony: Failed to download resource "symphony"
Download failed: https://github.com/coin-or-tools/homebrew-coinor/releases/download/symphony-5.6.18/symphony-5.6.18.x86_64_linux.bottle.tar.gz
```
The above was after I tried changing the `root_url`, since I thought that might matter, but the same thing happens if I leave the root URL pointing to the old version.

**Comment by dpo** [comment]
Looks like it worked. But you're really supposed to leave the `bottle` section in there.

**Comment by tkralphs** [comment]
> But you're really supposed to leave the bottle section in there.

OK, but how? What am I doing wrong? I don't see any way to leave it in there.

**Comment by dpo** [comment]
Just don't remove it from the file! :-)

**Comment by tkralphs** [comment]
> Just don't remove it from the file! :-)

Yes, but I described what happens if I do that. Am I just not supposed to do any testing locally, just change the URL, push, and hope for the best? The update of the source SHA and everything else happens automagically? It seems weird not to build locally first and with the `bottle do` part, I simply don't seem to be able to do that. Maybe I am missing an installation option for forcing it not to install bottles? I'm sure you are doing things differently than me in some small way and I just can't put my finger on what it is.

**Comment by dpo** [comment]
Locally, brew doesn't know that the bottles in the bottle section aren't the latest when you change the version. You want to install from source with `brew install -s`. But there's no need to touch the bottle section in your PR. The bot will update it if the build goes through.

**Pull request closed without merging**